src/Swagger/Serializer/DocumentationNormalizer.php

```diff
@@ -785,12 +785,14 @@ private function flattenMimeTypes(array $responseFormats): array
     private function getLinkObject(string $resourceClass, string $operationId, string $path): array
     {
         $linkObject = $identifiers = [];
+        $resourceMetadataAttributes = $this->resourceMetadataFactory->create($resourceClass)->getAttributes();
         foreach ($this->propertyNameCollectionFactory->create($resourceClass) as $propertyName) {
             $propertyMetadata = $this->propertyMetadataFactory->create($resourceClass, $propertyName);
             if (!$propertyMetadata->isIdentifier()) {
                 continue;
             }
 
+            $propertyName = $resourceMetadataAttributes[$propertyName]['serialized_name'] ?? $propertyName;
```

Comment:
I don't understand why this attribute would exist. ResourceMetadata attributes have nothing to do with Serializer's metadata.

I don't think that it's the good way to fix this. A better way would be to copy the Serializer's `serialized_name` into account in the property metadata and to use it everywhere: https://github.com/api-platform/core/blob/master/src/Metadata/Property/Factory/SerializerPropertyMetadataFactory.php